File: truss-compiler/src/main/java/truss/compiler/preprocessor/ElIfDirective.java
package truss.compiler.preprocessor;

class ElIfDirective implements Directive {
    private final boolean expression;

    public ElIfDirective(boolean expression) {
        this.expression = expression;
    }

    public boolean isExpression() {
        return expression;
    }

    @Override
    public DirectiveKind getKind() {
        return DirectiveKind.ELIF;
    }
}

File: Truss.Compiler/PreProcessor/ElIfDirective.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace Truss.Compiler.PreProcessor {
    internal class ElIfDirective : IDirective {
        public ElIfDirective(bool expression) {
            Expression = expression;
        }

        public bool Expression { get; private set; }

        public DirectiveKind Kind {
            get { return DirectiveKind.ElIf; }
        }
    }
}
